monai/data/image_reader.py

```diff
@@ -226,64 +238,53 @@ def _get_meta_dict(self, img) -> Dict:
 
         """
         img_meta_dict = img.GetMetaDataDictionary()
-        meta_dict = {}
-        for key in img_meta_dict.GetKeys():
-            # ignore deprecated, legacy members that cause issues
-            if key.startswith("ITK_original_"):
-                continue
-            if (
-                key == "NRRD_measurement frame"
-                and int(itk.Version.GetITKMajorVersion()) == 5
-                and int(itk.Version.GetITKMinorVersion()) < 2
-            ):
-                warnings.warn(
-                    "Ignoring 'measurement frame' field. "
-                    "Correct reading of NRRD05 files requires ITK >= 5.2: `pip install --upgrade --pre itk`"
-                )
-                continue
-            meta_dict[key] = img_meta_dict[key]
-        meta_dict["origin"] = np.asarray(img.GetOrigin())
+        meta_dict = {key: img_meta_dict[key] for key in img_meta_dict.GetKeys() if not key.startswith("ITK_")}
+
         meta_dict["spacing"] = np.asarray(img.GetSpacing())
-        meta_dict["direction"] = itk.array_from_matrix(img.GetDirection())
         return meta_dict
 
     def _get_affine(self, img):
         """
         Get or construct the affine matrix of the image, it can be used to correct
         spacing, orientation or execute spatial transforms.
-        Construct Affine matrix based on direction, spacing, origin information.
-        Refer to: https://github.com/RSIP-Vision/medio
 
         Args:
-            img: a ITK image object loaded from a image file.
+            img: an ITK image object loaded from an image file.
 
         """
         direction = itk.array_from_matrix(img.GetDirection())
         spacing = np.asarray(img.GetSpacing())
         origin = np.asarray(img.GetOrigin())
 
         direction = np.asarray(direction)
-        affine: np.ndarray = np.eye(direction.shape[0] + 1)
-        affine[(slice(-1), slice(-1))] = direction @ np.diag(spacing)
-        affine[(slice(-1), -1)] = origin
+        sr = min(max(direction.shape[0], 1), 3)
+        affine: np.ndarray = np.eye(sr + 1)
+        affine[:sr, :sr] = direction[:sr, :sr] @ np.diag(spacing[:sr])
+        affine[:sr, -1] = origin[:sr]
+        flip_diag = [[-1, 1], [-1, -1, 1], [-1, -1, 1, 1]][sr - 1]
+        affine = np.diag(flip_diag) @ affine
         return affine
 
     def _get_spatial_shape(self, img):
         """
-        Get the spatial shape of image data, it doesn't contain the channel dim.
+        Get the spatial shape of `img`.
 
         Args:
-            img: a ITK image object loaded from a image file.
+            img: an ITK image object loaded from an image file.
 
         """
-        # the img data should have no channel dim or the last dim is channel
-        shape = list(itk.size(img))
-        shape.reverse()
-        return np.asarray(shape)
+        # the img data should have no channel dim
+        sr = min(img.__dict__.get("dim[0]", 3), 3)
```

Comment:
I don't think "dim[0]" would ever be a valid member of an object's `__dict__` unless it was explicitly added. Is this something ITK adds?

Reply:
Hi @thewtex could you please help here, How to determine the number of spatial dimensions of an itk image? Here I'm trying to use an attribute that I found when the input is a nifiti file.

Reply:
Hi @ericspod  I've checked that `img.keys()` has the ` 'dim[0]'`,
and `img['dim[0]']` works, but the syntax donesn't look right...

I'm revising this to `sr = itk.array_from_matrix(img.GetDirection()).shape[0]`, we could update this later